lib/r10k/module/base.rb

```diff
@@ -63,7 +64,9 @@ def full_path
   # Synchronize this module with the indicated state.
   # @abstract
   def sync(opts={})
-    raise NotImplementedError
+    spec_dir = File.join(path, 'spec')
```

Comment:
This may change the scope of this PR, but would it be possible to make this a configuration option that defaults to `spec`? It would be nice to have a module blocklist that can allow users to specify which directories will be deleted. I could see adding additional items such as `vendor`, or `.bundle` to this list.

Reply:
yeah, that's definitely an idea that's been floating around in my head. A community member also brought up wanting to exclude other files. You're right though, it does change the scope of this work though, so I'll need to talk to Product about this. /cc @npwalker @puppetlabs/puppetserver-maintainers